Documentation/network/servicemesh/mutual-authentication/mutual-authentication.rst

```diff
@@ -0,0 +1,116 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _gs_mutual_authentication:
+
+*********************
+Mutual Authentication
+*********************
+
+.. include:: ../../../beta.rst
+
+Mutual TLS Requirements
+#######################
+
+Mutual Transport Layer Security (MTLS) is a mechanism that ensures the authenticity, integrity, and confidentiality of data exchanged between two entities over a network.
+
+Unlike traditional TLS, which involves a one-way authentication process where the client verifies the server's identity, mutual TLS adds an additional layer of security by requiring both the client and the server to authenticate each other.
+
+Mutual TLS is also known as two-way TLS. Let's remind ourselves the goals behind TLS.
+
+TLS Goals
+---------
+
+`RFC-8446 <https://datatracker.ietf.org/doc/html/rfc8446#section-1>`_ defines the goals of a TLS secure channel as having the following properties:
+
+- Authentication: The server side of the channel is always authenticated; the client side is optionally authenticated.
+  Authentication can happen via asymmetric cryptography (e.g., RSA [RSA], the Elliptic Curve Digital Signature Algorithm (ECDSA)
+  [ECDSA], or the Edwards-Curve Digital Signature Algorithm (EdDSA) [RFC8032]) or a symmetric pre-shared key (PSK).
+
+- Confidentiality: Data sent over the channel after establishment is only visible to the endpoints. TLS does not hide the length of
+  the data it transmits, though endpoints are able to pad TLS records in order to obscure lengths and improve protection against
+  traffic analysis techniques.
+
+- Integrity: Data sent over the channel after establishment cannot be modified by attackers without detection.
+
+In traditional TLS transactions, these are handled by having a public key handshake performed (which satisfies the *Authentication* requirement), which then negotiates a symmetric key that is used by a record protocol to encrypt the traffic in transit (which satisfies the *Confidentiality* and *Integrity* properties). 
+
+MTLS in Cilium
```

Comment:
I think that we should avoid the use of the term "mTLS" except for in "mTLS-based mutual authentication".

```suggestion
Mutual Authentication in Cilium
```